core/shared/src/main/scala/org/typelevel/log4cats/AskStructuredLogger.scala

```diff
@@ -0,0 +1,71 @@
+package org.typelevel.log4cats
+
+import cats.Monad
+import cats.mtl.Ask
+
+private[log4cats] class AskStructuredLogger[F[_]](logger: StructuredLogger[F])(implicit
+    F: Monad[F],
+    ask: Ask[F, Map[String, String]]
```

Comment:
The `Map` returned here could be a view (say, from otel4s), but it needs to have a fast `++` to another `Map`.